### ISSUE TITLE
Could not load GPU driver

### ISSUE THREAD

**Issue by Wantoast** [Bumblebee-Project/Bumblebee#613]
I've just come across another issue about the same error I'm having and I wasn't sure about opening a new one.
Still, I think my problem might be different, and in that case having a separate thread would be less confusing.

So, I've installed `bumblebee-nvidia` and `primus` following [this guide](https://wiki.debian.org/Bumblebee), and when I tried `optirun glxgears` I got:
```
[ 2869.884806] [ERROR]Cannot access secondary GPU - error: Could not load GPU driver

[ 2869.884928] [ERROR]Aborting because fallback start is disabled.
```
These are my current settings (without comments):
```
[bumblebeed]
VirtualDisplay=:8
KeepUnusedXServer=false
ServerGroup=bumblebee
TurnCardOffAtExit=false
NoEcoModeOverride=false
Driver=nvidia
XorgConfDir=/etc/bumblebee/xorg.conf.d


[optirun]
Bridge=auto
VGLTransport=proxy
PrimusLibraryPath=/usr/lib/x86_64-linux-gnu/primus:/usr/lib/i386-linux-gnu/primus:/usr/lib/primus:/usr/lib32/primus
AllowFallbackToIGC=false

[driver-nvidia]
KernelDriver=nvidia-current
PMMethod=auto
LibraryPath=/usr/lib/x86_64-linux-gnu/nvidia:/usr/lib/i386-linux-gnu/nvidia:/usr/lib/nvidia
XorgModulePath=/usr/lib/nvidia,/usr/lib/xorg/modules
XorgConfFile=/etc/bumblebee/xorg.conf.nvidia

[driver-nouveau]
KernelDriver=nouveau
PMMethod=auto
XorgConfFile=/etc/bumblebee/xorg.conf.nouveau
```

```
Section "ServerLayout"
    Identifier  "Layout0"
    Option      "AutoAddDevices" "false"
    Option      "AutoAddGPU" "false"
EndSection

Section "Device"
    Identifier  "DiscreteNvidia"
    Driver      "nvidia"
    VendorName  "NVIDIA Corporation"
    BusID "PCI:01:00:0"
    Option "NoLogo" "true"
    Option "UseEDID" "false"
    Option "UseDisplayDevice" "none"
EndSection
```
As you can see, I've already tried to use `KernelDriver=nvidia-current`, `Driver=nvidia` as well as specifying the BusID, which I got with `lspci | egrep 'VGA|3D'`.
I also tried to manually modprobe nvidia-current, with this output:
```
ERROR: could not insert 'nvidia_current': Unknown symbol in module, or unknown parameter (see dmesg)
```
So I checked dmesg (full paste [here](http://pastebin.com/RBQFCuUT) if needed):
```

[ 1651.672152] bbswitch: enabling discrete graphics
[ 1652.124057] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
[ 1665.397107] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
[ 1670.281192] bbswitch: disabling discrete graphics
[ 1670.281208] ACPI Warning: \_SB_.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20140724/nsarguments-95)
[ 1671.808955] bbswitch: enabling discrete graphics
[ 1672.256008] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
[ 1941.422061] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
[ 1956.089715] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
[ 2868.430205] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
[ 3450.016618] bbswitch: disabling discrete graphics
[ 3450.016637] ACPI Warning: \_SB_.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20140724/nsarguments-95)
[ 3452.807518] bbswitch: enabling discrete graphics
[ 3453.254496] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
[ 4228.249387] nvidia: Unknown symbol acpi_os_wait_events_complete (err 0)
```

Also:
```
wantoast@crunchbang:~$ lspci | grep VGA
00:02.0 VGA compatible controller: Intel Corporation 2nd Generation Core Processor Family Integrated Graphics Controller (rev 09)
01:00.0 VGA compatible controller: NVIDIA Corporation GF119 [GeForce GT 520M] (rev a1)
```
Lastly, using nouveau instead of the proprietary driver works fine.

What's the problem here?
Thanks in advance.

**Comment by ArchangeGabriel** [comment]
Sorry for the lack of answer so far. That’s an issue in old NVIDIA drivers, it appeared because of a changes in Kernel 3.13, but has been fixed on nvidia side since. I suspect you’re running a Kernel>=3.13 with an old NVIDIA driver, please consider upgrading this one.

**Comment by stevelacy** [comment]
I am having the same problem on 3.13 GTX 970m.
Trying not to piggyback on his issue, @ArchangeGabriel can you state which driver is not outdated?

**Comment by stevelacy** [comment]
Update:
I have a working (not yet CUDA) install:
```
Linux mint 17.1 3.13
GTX 970m
```

Repos added:
```
sudo add-apt-repository ppa:bumblebee/stable
sudo add-apt-repository ppa:xorg-edgers/ppa
```

Installed packages:
```
sudo apt-get install bumblebee-nvidia primus
sudo apt-get install nvidia-346 nvidia-346-uvm
```

Restarted bumblebee and optirun works well.

**Comment by ArchangeGabriel** [comment]
@stevelacy I would have, but I’m not sure of which ones (especially since it might have been fixed in 304.xx but not in 331.yy for instance, since nvidia manages several branches in parallel), and don’t have the time to check all changelogs for this.

Anyway, since you confirm this is solved by upgrading, I’m closing this.

**Comment by stevelacy** [comment]
Another update:
CUDA (Blender 2.7.x) works well with the aforesaid nvidia drivers, you just need to run it as root.
